### ISSUE TITLE
Restrict appointments in the future, set arbitrary periods of availability. 

### ISSUE THREAD

**Issue by maxlandry** [alextselegidis/easyappointments#567]
Hi!

First of all, thank you for your great work.

It would be great if we could automatically restrict appointment in the future and only allow booking for 1 or 2 weeks at a time. For example, if we are December 1st, I don't want my customer to be able to book an appointment for January, but rather for the next 1 or 2 weeks of December, but as we move closer to January, I start to open new slots for this month.

**Comment by alciomarhollanda** [comment]
it would be really good

**Comment by alextselegidis** [comment]
@maxlandry This is a good feature request, marked for development. 

<small>
<table>
<tr>
<td><img src="http://easyappointments.org/img/easyappointments-signature.png"></td>
<td>&nbsp;&nbsp;Alex Tselegidis, Easy!Appointments Creator <br>
&nbsp;&nbsp;<a href="http://alextselegidis.com">Need a customization? Contact me in person!</a></td>
</tr>
</table>
</small>

**Comment by KibGim** [comment]
@alextselegidis Are there any updates?
It would indeed be very helpful in many usecases!
Could be something handled like **max_date** here https://easy-appointments.net/documentation/

@maxlandry @alciomarhollanda : Did you solve somehow?

Thanks

**Comment by alextselegidis** [comment]
Hello! 

This feature is actually part of v1.4.0, where users will be able to set working plan exceptions on specific days. 

With this feature you can disable the original working plan and only define the exact days the provider is going to work. 

For this I will close this ticket as duplicate. 

<small>
<table>
<tr>
<td><img src="https://easyappointments.org/img/easyappointments-signature.png"></td>
<td>&nbsp;&nbsp;Alex Tselegidis, Easy!Appointments Creator <br>
&nbsp;&nbsp;<a href="https://alextselegidis.com">Need a customization? Contact me in person!</a></td>
</tr>
</table>
</small>

**Comment by KibGim** [comment]
Thanks @alextselegidis for the great feedback! 
From my previous post about **max_date**:
> ..you can set time formula like set default day 5 days from now +5d or next week +1w etc.

Will this also be present? (It is what I am searching for, and equals what @maxlandry described).
Thanks

**Comment by popod** [comment]
@alextselegidis I think that this is not covered by v1.4.0 (or which is the duplicate issue) ?

Here we want to define a maximum limit after which users cannot book an appointment.. For exemple this is useful if I don't want my customers to book an appointment more that 2 months after now.

**Comment by alextselegidis** [comment]
@popod Technically you can just set working plan exceptions for the next couple of weeks which means that no one can book after that. Let us keep it this way cause the availability management will be drastically change in the next releases, as I want to add support for even more sophisticated scenarios. 

<small>
<table>
<tr>
<td><img src="https://easyappointments.org/img/easyappointments-signature.png"></td>
<td>&nbsp;&nbsp;Alex Tselegidis, Easy!Appointments Creator <br>
&nbsp;&nbsp;<a href="https://alextselegidis.com">Need a customization? Contact me in person!</a></td>
</tr>
</table>
</small>

**Comment by zefkide** [comment]
Hello, I'm using latest stable release 1.4.3 and trying to limit the booking calendar to max 30 days in advance.
I tried unsuccessfully to modify assets/js/frontend_book.js by adding maxDate parameter as :
```
        $('#select-date').datepicker({
            dateFormat: 'dd-mm-yy',
            firstDay: weekDayId,
            minDate: 0,
            **maxDate: '+30d',**
            defaultDate: Date.today(),
```
It doesn't work. Can this issue be handled without using working plan exceptions for all the resting days ? :)
Thank you in advance for your answer,

**Comment by popod** [comment]
@alextselegidis sadly no : working plan exceptions do not cover this issue request.

The main request of this issue is to restrict public booking appointment from now to (for example) one month in the future. This limit (the date in one month) is not static and should be dynamic.. This "one month" windows should be automatically maintained tomorrow, in one week, ... to always allows people to book an appointment between the current date and a month in the future.

Working plan exceptions does not cover this use case because by using them, each day, it is needed du create a new working exception to open a news day for booking..

This will be really useful to allow the user to define that the booking is enabled up to two months (for example) in the future and let the code logic to automatically manage the open days for a public booking.

Hop this is clear and it helps.

**Comment by zefkide** [comment]
This should be a bug rather than a feature. No offense but no one need appointments for now to 200 years in future or whatever. The open period range for appointments should be a UI parameter or at least a variable in code.

Finally my solution worked after a firefox cache clear. The front calendar is locked now to max 30 days in advance grace to this line:
`maxDate: '+30d',`

Otherwise great piece of opensource code, thanks guys

**Comment by popod** [comment]
@alextselegidis I've implemented something like that : we can define a booking window during which the customers can book an appointment.

For exemple, with a booking window of 90 days, customers can only book an appointment from now to now + 90 days. This windows is dynamic and change days after days to always allow to book appointments from now to now + 90 days.

Are you interested by a PR ?

**Comment by alextselegidis** [comment]
Hello @popod, 

Thanks for your message and proposal, but this feature is already implemented and will be part of the 1.5 release. 

You can already check it on the latest dev package. 

<small>
<table>
<tr>
<td><img src="https://easyappointments.org/img/easyappointments-signature.png" style="margin-right: 5px"></td>
<td style="line-height: 2">Alex Tselegidis, Easy!Appointments Creator <br><a href="https://easyappointments.org/get-a-free-quote">Need a customization? Get a free quote!</a></td>
</tr>
</table>
</small>
